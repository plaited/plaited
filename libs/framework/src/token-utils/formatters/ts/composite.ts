import { camelCase } from '@plaited/utils'
import { Formatter, CompositeToken } from '../../types.js'
import { hasAlias } from '../has-alias.js'
import { resolveTSVar, isValidAlias } from '../ts-utils.js'

export const composite: Formatter<CompositeToken> = (token, { tokenPath, allTokens }) => {
  const { $value } = token
  if (hasAlias($value) && !isValidAlias($value, allTokens)) {
    return ''
  }
  if (hasAlias($value)) {
    return `export const ${camelCase(tokenPath.join(' '))} = ${resolveTSVar($value, allTokens)}`
  }
  const toRet: string[] = []
  for (const key in $value) {
    const val = $value[key]
    if (isValidAlias(val, allTokens)) {
      toRet.push(`  ${key}: ${resolveTSVar(val, allTokens)},`)
    }
  }
  return [`export const ${camelCase(tokenPath.join(' '))} = {`, ...toRet, '}'].join('\n')
}
