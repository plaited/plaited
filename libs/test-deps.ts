// -- std --
export {
  assert,
  assertEquals,
  assertFalse,
} from 'https://deno.land/std@0.176.0/testing/asserts.ts'
export { assertSnapshot } from 'https://deno.land/std@0.176.0/testing/snapshot.ts'
export {
  afterEach,
  beforeEach,
  describe,
  it,
} from 'https://deno.land/std@0.176.0/testing/bdd.ts'
export { resolve } from 'https://deno.land/std@0.175.0/path/mod.ts'
// -- sinon --
export { default as sinon } from 'https://esm.sh/sinon@15.0.1'

// -- beautify --
export { default as beautify } from 'https://esm.sh/beautify@0.0.8'
