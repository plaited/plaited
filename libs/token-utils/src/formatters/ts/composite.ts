import { camelCase } from '@plaited/utils'
import { Formatter, AliasValue, FlexToken, GridToken, TypographyToken } from '../../types.js'
import { hasAlias } from '../has-alias.js'
import { resolveTSVar } from '../ts-utils.js'

type CompositeToken = FlexToken | GridToken | TypographyToken

export const composite: Formatter<CompositeToken> = (token, { tokenPath, allTokens }) => {
  const { $value } = token
  if (hasAlias($value)) {
    return `export const ${camelCase(tokenPath.join(' '))} = ${resolveTSVar($value as AliasValue, allTokens)}`
  }
  const toRet = Object.entries($value).map(([key, val]) => `  ${key}: ${resolveTSVar(val, allTokens)},`)
  return [`export const ${camelCase(tokenPath.join(' '))} = {`, ...toRet, '}'].join('\n')
}
