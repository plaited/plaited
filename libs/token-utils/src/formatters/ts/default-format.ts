import { camelCase } from '@plaited/utils'
import { Formatter } from '../../types.js'
import { hasAlias } from '../has-alias.js'
import { isValidAlias, getCssVar } from '../ts-utils.js'

export const defaultFormat: Formatter = (token, { allTokens, tokenPath }) => {
  const { $value } = token
  if (hasAlias($value) && !isValidAlias($value, allTokens)) {
    return ''
  }
  return `export const ${camelCase(tokenPath.join(' '))} = ${getCssVar(tokenPath)}`
}
