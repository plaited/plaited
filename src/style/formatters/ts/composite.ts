import type { CompositeToken } from '../../token.types.js'
import type { Formatter } from '../formatters.types.js'
import { camelCase } from '../../../utils/case.js'
import { hasAlias } from '../has-alias.js'
import { resolveTSVar, isValidAlias } from '../ts-utils.js'

export const composite: Formatter<CompositeToken> = (token, { tokenPath, allTokens }) => {
  const { $value } = token
  if (hasAlias($value) && !isValidAlias($value, allTokens)) {
    return ''
  }
  if (hasAlias($value)) {
    return `export const ${camelCase(tokenPath.join(' '))} = ${resolveTSVar($value, allTokens)}`
  }
  const toRet: string[] = []
  for (const key in $value) {
    const val = $value[key]
    if (isValidAlias(val, allTokens)) {
      toRet.push(`  ${key}: ${resolveTSVar(val, allTokens)},`)
    }
  }
  return [`export const ${camelCase(tokenPath.join(' '))} = {`, ...toRet, '}'].join('\n')
}
