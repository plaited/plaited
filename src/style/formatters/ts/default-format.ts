import type { Formatter } from '../formatters.types.ts'
import { camelCase } from '../../../utils/case.ts'
import { hasAlias } from '../has-alias.ts'
import { isValidAlias, getCssVar } from '../ts-utils.ts'

export const defaultFormat: Formatter = (token, { allTokens, tokenPath }) => {
  const { $value } = token
  if (hasAlias($value) && !isValidAlias($value, allTokens)) {
    return ''
  }
  return `export const ${camelCase(tokenPath.join(' '))} = ${getCssVar(tokenPath)}`
}
