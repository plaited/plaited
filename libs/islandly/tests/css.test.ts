import { assertSnapshot } from '../../test-deps.ts'
import { css } from '../mod.ts'

Deno.test('css()', async (t) => {
  await assertSnapshot(
    t,
    css`
    .button {
      background-color: rgba(255, 0, 0, 0.2);
    }
    .button4:disabled {
      background-color: rgba(255, 0, 0, 0.8);
    }
    ._button-4:disabled {
      background-color: rgba(00, 50, 25, 0.8);
    }
    ._button-4:disabled + :not(.link){
      background-color: rgba(00, 50, 25, 0.8);
    }
    `,
  )
  await assertSnapshot(
    t,
    css`
    .button {
      background-color: rgba(255, 0, 0, 0.2);
    }
    .button4:disabled {
      background-color: rgba(255, 0, 0, 0.8);
    }
    ._button-4:disabled  > .combinator {
      background-color: va(--background-color);
    }
    ._button-4:disabled + :not(.link){
      background-color: var(--background-color-disabled);
    }
    `,
  )
})
