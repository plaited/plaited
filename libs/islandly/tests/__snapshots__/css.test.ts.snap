export const snapshot = {};

snapshot[`css() 1`] = `
{
  classes: {
    "_button-4": "_button-4_NzU3MD",
    button: "button_NzU3MD",
    button4: "button4_NzU3MD",
    link: "link_NzU3MD",
  },
  styles: ".button_NzU3MD { background-color: rgba(255, 0, 0, 0.2); } .button4_NzU3MD:disabled { background-color: rgba(255, 0, 0, 0.8); } ._button-4_NzU3MD:disabled { background-color: rgba(00, 50, 25, 0.8); } ._button-4_NzU3MD:disabled + :not(.link_NzU3MD){ background-color: rgba(00, 50, 25, 0.8); }",
}
`;

snapshot[`css() 2`] = `
{
  classes: {
    "_button-4": "_button-4_MjE0Mz",
    button: "button_MjE0Mz",
    button4: "button4_MjE0Mz",
    link: "link_MjE0Mz",
  },
  styles: ".button_MjE0Mz { background-color: rgba(255, 0, 0, 0.2); } .button4_MjE0Mz:disabled { background-color: rgba(255, 0, 0, 0.8); } ._button-4_MjE0Mz:disabled { background-color: va(--background-color); } ._button-4_MjE0Mz:disabled + :not(.link_MjE0Mz){ background-color: var(--background-color-disabled); }",
}
`;
